cmd/data.go

```diff
@@ -164,22 +165,48 @@ func (cmd *DataCmd) Execute(ctx context.Context, f *flag.FlagSet, _ ...interface
 			return subcommands.ExitFailure
 		}
 	}
+	streamingCfg := streaming.StreamingCfg{}
+	// Launch a stream if streaming profile found.
+	if sourceProfile.Ty == profiles.SourceProfileTypeStreaming {
+		streamingCfg, err = streaming.ReadStreamingConfig(sourceProfile.Streaming.MySQL.StreamingConfig, targetProfile.Conn.Sp.Dbname)
+		if err != nil {
+			err = fmt.Errorf("error reading streaming config: %v", err)
+			return subcommands.ExitFailure
+		}
+
+		err = streaming.LaunchStream(ctx, sourceProfile.Streaming.MySQL.Conn.Db, project, streamingCfg.DatastreamCfg)
```

Comment:
Does this return an error synchronously ?

Reply:
Yes, so this is our method which is synchronous. The async bits are followed by waits() in side this.